elyra/pipeline/processor_local.py

```diff
@@ -52,7 +63,13 @@ def process(self, pipeline):
         # Sort operations based on dependency graph (topological order)
         operations = LocalPipelineProcessor._sort_operations(pipeline.operations)
         for operation in operations:
-            self._execute_operation(operation)
+            filepath = self.get_absolute_path(operation.filename)
+
+            try:
+                operation_processor = self._operation_processor_registry[operation.classifier]
+                await operation_processor.process(operation, filepath)
```

Comment:
Seems like `operation` should be the only parameter to this method - especially since `filename` is derived from `operation.filename` and some future operations may not even be file-based.  The handling of `filepath` here just feels out of place and it should really be a function of the operation processor.

Reply:
the problem is with the root-dir used to create the absolute path

Reply:
I don't understand.  Every node/operation has the same `root-dir`, the computation of `filepath` will be the same inside or outside the `Operation` class.  If its because it doesn't have access to `root-dir` then can't `root-dir` be specified during construction?